src/packages/utils/src/things/json-rpc/json-rpc-quantity.ts

```diff
@@ -1,87 +1,135 @@
-import { bufferToBigInt, BUFFER_EMPTY } from "../../utils";
+import { bufferToBigInt } from "../../utils/buffer-to-bigint";
 import { BaseJsonRpcType } from "./json-rpc-base-types";
-// TODO(perf): rewrite this stuff since it isn't really caching anything
+import { JsonRpcInputArg } from "./input-parsers";
+import { BUFFER_EMPTY, BUFFER_ZERO } from "../../utils/constants";
+
 export class Quantity extends BaseJsonRpcType {
+  public static Empty = Quantity.from(BUFFER_EMPTY, true);
+  public static Zero = Quantity.from(BUFFER_ZERO);
+  public static One = Quantity.from(1);
+  public static Gwei = Quantity.from(1000000000);
+
+  private static ZERO_VALUE_STRING = "0x0";
+
   _nullable: boolean = false;
-  public static from(
-    value: number | bigint | string | Buffer,
-    nullable = false
-  ) {
+
+  public static from(value: JsonRpcInputArg | Quantity, nullable = false) {
     if (value instanceof Quantity) return value;
-    const q = new Quantity(value);
-    q._nullable = nullable;
-    return q;
+    return new Quantity(value, nullable);
+  }
+
+  constructor(value: JsonRpcInputArg, nullable?: boolean) {
+    super(value);
+    if (value === "0x") {
+      throw new Error('Cannot wrap "0x" as a json-rpc Quantity type; Quantity must contain at least one digit');
```

Comment:
In theory this shouldn't ever happen right? In any case, I don't think "digit" is the right word here, since it can be non-digits (if we are referring to `value` itself)

Reply:
It _shouldn't_ happen, but it _can_, this is a fix for https://github.com/trufflesuite/ganache/issues/2857

Agreed the message isn't perfect - what do you think of `Cannot wrap "0x" as a json-rpc Quantity type; strings must contain at least one hexadecimal symbol.`?

Reply:
I also added a test to target this specific validation case.

Reply:
Ah, that makes sense!

Does the word "character" read better?

```
Cannot wrap "0x" as a json-rpc Quantity type; strings must contain at least one hexadecimal character.
````